File: bug45_function_returns_constant/solved/main.java
public class main { // correct part
    static long hash_val(long v, long step) { // correct part
        long result = (v ^ step) * 0x9E3779B97F4A7C15L; // correct part
        result = (result << 13) | (result >>> 51);
        return result; // correct part
    }

    public static void main(String[] args) {
        long[] data = new long[10];
        for (int i = 1; i <= 10; i++) { // correct part
            data[i - 1] = 0x111L * i; // correct part
        } // correct part
        long acc = 0;
        for (int i = 0; i < 10; i++) {
            acc ^= hash_val(data[i], i); // correct part
            acc *= 0xBF58476D1CE4E5B9L; // correct part
        }
        System.out.println("0x" + Long.toHexString(acc)); // correct part
    }
}

File: bug45_function_returns_constant/solved/main.js
function hash_val(v, step) { // correct part
    let result = BigInt.asUintN(64, (v ^ step) * 0x9E3779B97F4A7C15n); // correct part
    return BigInt.asUintN(64, (result << 13n) | (result >> 51n)); // correct part
}

function run() { // correct part
    let data = []; // correct part
    for (let i = 1n; i <= 10n; i++) { // correct part
        data.push(0x111n * i); // correct part
    } // correct part
    let acc = 0n; // correct part
    for (let i = 0n; i < 10n; i++) { // correct part
        acc ^= hash_val(data[Number(i)], i); // correct part
        acc = BigInt.asUintN(64, acc * 0xBF58476D1CE4E5B9n); // correct part
    } // correct part
    console.log("0x" + acc.toString(16)); // correct part
}
run(); // correct part
